js/app/dashboard/redux/actions.js

```diff
@@ -309,26 +320,30 @@ export function modifyTaskList(username, tasks) {
 
       return {
         ...rt,
+        isAssigned: true,
         position,
       }
     })
 
     dispatch(modifyTaskListRequest(username, newTasks))
 
-    axios
-      .put(url, data, {
+    try {
+      var response =  await axios.put(url, data, {
         withCredentials: true,
         headers: {
           'Content-Type': 'application/ld+json'
         },
       })
-      .then(res => {
-        dispatch(modifyTaskListRequestSuccess(res.data))
-      })
-      .catch(error => {
-        // eslint-disable-next-line no-console
-        console.error(error)
-      })
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
+    }
+
+    // return a promise so we can chain after dispatching this function
+    return new Promise((resolve) => {
```

Comment:
There is no need to create a Promise explicitly. As it's an async function JS will wrap returned result into a promise

Reply:
Oh ok I didnt know that thx